support/oidc-discovery-provider/workload_api.go

```diff
@@ -109,59 +108,51 @@ func (s *WorkloadAPISource) pollEvery(ctx context.Context, conn *grpc.ClientConn
 	}
 }
 
-func (s *WorkloadAPISource) pollOnce(ctx context.Context, client workload_pb.SpiffeWorkloadAPIClient) {
-	// Ensure the stream gets cleaned up
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-
-	stream, err := client.FetchJWTBundles(ctx, &workload_pb.JWTBundlesRequest{})
+func (s *WorkloadAPISource) pollOnce(ctx context.Context, client *workloadapi.Client) {
+	jwtBundles, err := client.FetchJWTBundles(ctx)
 	if err != nil {
 		s.log.WithError(err).Warn("Failed to fetch JWKS from the Workload API")
 		return
 	}
 
-	resp, err := stream.Recv()
-	if err != nil {
-		if err == io.EOF {
-			s.log.Warn("Workload API stream closed before bundle received")
-			return
-		}
-		s.log.WithError(err).Warn("Failed to fetch JWKS from the Workload API")
+	jwtBundle, ok := jwtBundles.Get(s.trustDomain)
+	if !ok {
+		s.log.WithField("trust_domain_id", s.trustDomain.IDString()).Error("No bundle for trust domain in Workload API response")
```

Comment:
nit: may you use `telemetry.TrustDomainID`

Reply:
:+1: